daemon/stores/flows.py

```diff
@@ -1,92 +1,80 @@
         :param port_expose: port expose for container flow
+        :param params: json payload to be sent
         :param kwargs: keyword args
-        :return: response from mini-jinad"""
-        try:
-            r = requests.post(
-                url=f'{self.host}/{self._kind}',
-                params={'port_expose': port_expose},
-                json=self.params,
-            )
-            if r.status_code != requests.codes.created:
+        :return: response from mini-jinad
+        """
+        self._logger.debug(f'sending POST request to mini-jinad on {uri}/{self._kind}')
+        async with aiohttp.request(
+            method='POST',
+            url=f'{uri}/{self._kind}',
+            params={'port_expose': str(port_expose)},
+            json=params,
+        ) as response:
+            if response.status != HTTPStatus.CREATED:
                 raise Runtime400Exception(
-                    f'{self._kind.title()} creation failed \n{"".join(r.json()["body"])}'
+                    f'{self._kind.title()} creation failed: {response.json()}'
                 )
-            return r.json()
-        except requests.exceptions.RequestException as ex:
-            self._logger.error(f'{ex!r}')
-            raise Runtime400Exception(
-                f'{self._kind.title()} creation failed: {r.json()}'
-            )
+            return await response.json()
 
-    def update(
-        self,
-        id: DaemonID,
-        kind: UpdateOperation,
-        dump_path: str,
-        pod_name: str,
-        shards: int = None,
-    ) -> Dict:
-        """Sends `put` request to `mini-jinad` to execute a command on a Flow.
+    @raise_if_not_alive
+    async def _update(self, uri: str, params: Dict, **kwargs) -> Dict:
+        """Sends `PUT` request to `mini-jinad` to execute a command on a Flow.
 
-        :param id: flow id
-        :param kind: type of update command to execute (only rolling_update for now)
-        :param dump_path: the path to which to dump on disk
-        :param pod_name: pod to target with the dump request
-        :param shards: nr of shards to dump
-        :return: response from mini-jinad"""
-        try:
-            params = {
-                'kind': kind,
-                'dump_path': dump_path,
-                'pod_name': pod_name,
-                'shards': shards,
-            }
-            r = requests.put(
-                url=f'{self[id].metadata.host}/{self._kind}', params=params
-            )
+        :param uri: uri of mini-jinad
+        :param params: json payload to be sent
+        :param kwargs: keyword args
+        :return: response from mini-jinad
+        """
 
-            if r.status_code != requests.codes.ok:
+        self._logger.debug(f'sending PUT request to mini-jinad on {uri}/{self._kind}')
+        async with aiohttp.request(
+            method='PUT', url=f'{uri}/{self._kind}', params=params
+        ) as response:
+            response_json = await response.json()
```

Comment:
What happens if there is some issue with the json deserialization? Same goes for other cases. Does this work then here or throw an exception?

Reply:
That gets caught in `containers.py` exception handler and raised to the invoker. But our exception sending logic to the user is poor now. It needs quite some work and cannot be done in this PR